### ISSUE TITLE
Sometimes redtt loops when the parsing fails miserably.

### ISSUE THREAD

**Issue by favonia** [RedPRL/redtt#289]
This happened several times. The latest example is to change => to ->.

**Comment by favonia** [comment]
Concrete example: after updating the parser, `redtt load-file modal.red` seems to loop with the following message:
```
Entering directory '/Immortal/redtt'
path.red:3:8:ERROR
  bool.red:10:2:ERROR
  Encountered error:
    Failure("Datatype not found: bool")
[diagnostic]: spent 0.000000s acting on dimensions
[diagnostic]: Tm spent 0.000003s in close_var
[diagnostic]: Tm spent 0.000000s in open_var
[diagnostic]: Cx spent 0.000000s in checking equality
Refine spent 0.000181s in normalizing types
```

**Comment by jonsterling** [comment]
wow. that's remarkable.

**Comment by jonsterling** [comment]
I wonder if I have fixed this.

**Comment by favonia** [comment]
If it is fixed, it was probably fixed by #300.

**Comment by favonia** [comment]
Now parsing errors seem to always stop redtt.
